src/gate_detect/src/GateDetection.cpp

```diff
@@ -0,0 +1,330 @@
+                 std::back_inserter(horLines),
+                 [=](cv::Vec4i line1) {
+                     return abs((line1)[1] - (line1)[3]) < _lowHorThresh;
+                 });
+
+    return horLines;
+}
+
+std::vector<PoleDetection>
+GateDetection::findVertPoles(std::vector<cv::Vec4i> vertLines) {
+    std::vector<PoleDetection> verticalPoles;
+
+    std::vector<cv::Vec4i>::iterator line1;
+    std::vector<cv::Vec4i>::iterator line2;
+
+    if (vertLines.size() <= 1)
+        return verticalPoles;
+
+    else if (vertLines.size() == 2) {
+        verticalPoles.push_back(
+        PoleDetection(vertLines.at(0), vertLines.at(1), m_Hor, b_Hor));
+
+        return verticalPoles;
+    }
+
+    for (line1 = vertLines.begin(); line1 != vertLines.end(); line1++) {
+        for (line2 = std::next(line1); line2 != vertLines.end();) {
+            if ((abs((*line1)[0] - (*line2)[0]) < 2) || (*line1 == *line2)) {
+                line2 = vertLines.erase(line2);
+
+            } else if (abs((*line1)[0] - (*line2)[0]) < _poleMax) {
+                PoleDetection verticalPole =
+                PoleDetection(*line1, *line2, m_Vert, b_Vert);
+
+                verticalPoles.push_back(verticalPole);
+
+                line2 = vertLines.erase(line2);
+
+            } else {
+                line2++;
+            }
+        }
+    }
+
+    return verticalPoles;
+}
+
+std::vector<PoleDetection>
+GateDetection::findHorPoles(std::vector<cv::Vec4i> horLines) {
+    std::vector<PoleDetection> horizontalPoles;
+
+    if (horLines.size() <= 1)
+        return horizontalPoles;
+
+    else if (horLines.size() == 2) {
+        horizontalPoles.push_back(
+        PoleDetection(horLines.at(0), horLines.at(1), m_Hor, b_Hor));
+
+        return horizontalPoles;
+    }
+
+    std::vector<cv::Vec4i>::iterator line1;
+    std::vector<cv::Vec4i>::iterator line2;
+
+    if (horLines.size() == 2) {
+        PoleDetection verticalPole = PoleDetection(
+        *(horLines.begin()), *(horLines.begin()++), m_Vert, b_Vert);
+
+        horizontalPoles.push_back(verticalPole);
+
+        return horizontalPoles;
+    }
+
+    for (line1 = horLines.begin(); line1 != horLines.end(); line1++) {
+        for (line2 = std::next(line1); line2 != horLines.end();) {
+            if ((abs((*line1)[1] - (*line2)[1]) < 2) || (*line1 == *line2)) {
+                line2 = horLines.erase(line2);
+
+            } else if (abs((*line1)[1] - (*line2)[1]) < _poleMax) {
+                PoleDetection verticalPole =
+                PoleDetection(*line1, *line2, m_Vert, b_Vert);
+
+                horizontalPoles.push_back(verticalPole);
+
+                line2 = horLines.erase(line2);
+
+            } else {
+                line2++;
+            }
+        }
+    }
+
+    return horizontalPoles;
+}
+
+std::vector<float>
+GateDetection::getGateVector(std::vector<PoleDetection> vertPoles,
+                             std::vector<PoleDetection> horPoles) {
+    std::vector<float> gateCoord(9, 0.0f);
```

Comment:
I think we should definitly make a `GateCoordinates` struct instead of just using a vector of floats, `gate_coordinates.lower_left_corner_x` is a lot easier to read then `gateCoordinates[3]`. This also helps prevent issues like putting things in the vector in the wrong order, or making the vector too long or short.

Reply:
Made gateVector into a struct